src/utils.rs

```diff
@@ -106,6 +100,8 @@ pub fn find_package_manifest(path: &Path, name: Option<&str>) -> Result<(PathBuf
         if match_package_name(&manifest, name) {
             return Ok((manifest_path, manifest));
         }
+        // TODO: The above always returns true if no name is set. Should we still keep scanning upwards
+        // when a name is set, and we encountered a Cargo.toml, but it did not contain a package by that name?
```

Comment:
Local testing shows that this is indeed wrong behaviour.

Reply:
That also matches the `` Recursively walk up the directories until finding a `Cargo.toml` `` docs more closely, too.